anomalib/models/draem/augmenter.py

```diff
@@ -0,0 +1,150 @@
+"""Augmenter module to generates out-of-distribution samples for the DRAEM implementation."""
+
+# Original Code
+# Copyright (c) 2022 VitjanZ
+# https://github.com/VitjanZ/DRAEM.
+# SPDX-License-Identifier: MIT
+#
+# Modified
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import glob
+import random
+from typing import Optional, Tuple
+
+import cv2
+import imgaug.augmenters as iaa
+import numpy as np
+import torch
+from torch import Tensor
+
+from anomalib.models.draem.perlin import rand_perlin_2d_np
+
+
+class Augmenter:
+    """Class that generates noisy augmentations of input images.
+
+    Args:
+        anomaly_source_path (Optional[str]): Path to a folder of images that will be used as source of the anomalous
+        noise. If not specified, random noise will be used instead.
+    """
+
+    def __init__(self, anomaly_source_path: Optional[str] = None):
+
+        if anomaly_source_path is not None:
+            self.anomaly_source_paths = sorted(glob.glob(anomaly_source_path + "/**/*.jpg", recursive=True))
```

Comment:
Not sure if I understand this. Why does this search for jpg images?

Reply:
See my comment in the other thread. I'll make this more flexible by adding different image extensions.